assets/queries/cloudFormation/sage_maker_without_vpc/query.rego

```diff
@@ -0,0 +1,21 @@
+package Cx
+
+CxPolicy [ result ] {
+   document := input.document
+   resource = document[i].Resources[name]
+   resource.Type == "AWS::SageMaker::NotebookInstance"
+   
+   not check_resources_type(document[i].Resources)
+   
+      result := {
+                "documentId": 		input.document[i].id,
+                "searchKey":        sprintf("Resources.%s", [name]),
+                "issueType":		   "MissingAttribute",
+                "keyExpectedValue": sprintf("Resources.%s has VPC", [name]),
+                "keyActualValue": 	sprintf("Resources.%s has not VPC", [name])
+              }
+}
+
+check_resources_type(resource) {
+   resource[_].Type == "AWS::EC2::VPC"
+}
```

Comment:
I think a better approach would be checking if the field `Properties.DirectInternetAccess` is defined and set to "Disabled" (default is "Enabled"). This forces the notebook instance to access resources only through VPC.